instance/tests/integration/integration_instance.py

```diff
@@ -79,6 +81,41 @@ def assert_swift_container_provisioned(self, instance):
         stat_result = stat_container(instance.swift_container_name)
         self.assertEqual(stat_result.read_acl, '.r:*')
 
+    def assert_mysql_db_provisioned(self, instance):
+        """
+        Verify that the MySQL database for the instance has been provisioned and can be
+        connected to with the credentials the instance provides.
+        """
+        for database in instance.mysql_databases:
+            connection = mysql.connect(
+                host=settings.INSTANCE_MYSQL_URL_OBJ.hostname,
+                user=database['user'],
+                passwd=instance._get_mysql_pass(database['user']),
+                port=settings.INSTANCE_MYSQL_URL_OBJ.port or 3306,
+            )
+            database_name = connection.escape_string(database['name']).decode()
+            cur = connection.cursor()
+            result = cur.execute(
+                "SELECT SCHEMA_NAME FROM INFORMATION_SCHEMA.SCHEMATA WHERE SCHEMA_NAME = '{db_name}'".format(
+                    db_name=database_name
+                )
+            )
+            self.assertEqual(result, 1)
+
+    def assert_mongo_db_provisioned(self, instance):
+        """
+        Verify that the databases ostensibly created by the provisioning process actually were.
+        """
+        mongo = pymongo.MongoClient(settings.INSTANCE_MONGO_URL)
+        db_names = mongo.database_names()
+        # This looks a little odd - basically, Mongo's `database_names` command doesn't list
+        # databases that don't have any documents in them. And, we have at least one database
+        # (forums) that doesn't have any documents inserted during the provisioning. So, as a
+        # result, if we require every db_name to be present, we'll fail, even though the db
+        # is, in actuality, there. So, we're checking that at least one of the databases that
+        # we're supposed to create shows up.
```

Comment:
Yeah, the "current database name" in mongo is treated more like an implicit label that gets attached to documents than some sort of structure that exists independently, so it doesn't really make sense to speak of a database name existing or not; from a behavioral perspective, there are only either documents annotated with a particular "database name" or not, and arbitrary new database names can be used at any time.

So I think all that we really need to do is test that the connection string/URL is valid; this seems like a reasonable way of doing that, as long as we are sure that the provisioning process always creates at least one mongo document.

Question: If we provision an instance without any demo courses, do mongo documents still get created? Is there any chance some future type of instance may start with no mongo documents at all (blank slate)?